### PR TITLE
GO-2767 Fix table pasting

### PR DESCRIPTION
**Pull request by fat-fellow**
---

- [ ] I understand that contributing to this repository will require me to agree with the [CLA](https://github.com/anyproto/.github/blob/main/docs/CLA.md)

---

### What type of PR is this? (check all applicable)

- [ ] 🍕 Feature
- [ ] 🐛 Bug Fix
- [ ] 📝 Documentation Update
- [ ] 🎨 Style
- [ ] 🧑‍💻 Code Refactor
- [ ] 🔥 Performance Improvements
- [ ] ✅ Test
- [ ] 🤖 Build
- [ ] 🔁 CI

### Added tests?

- [ ] 👍 yes
- [ ] 🙅 no, because they aren't needed
- [ ] 🙋 no, because I need help

### Added to documentation?

- [ ] 📜 README.md
- [ ] 📓 [tech-docs](https://github.com/anyproto/tech-docs)
- [ ] 🙅 no documentation needed

## Summary by CodeRabbit

- **New Features**
	- Enhanced pasting functionality to support pasting content directly into table cells.

- **Bug Fixes**
	- Corrected a typo in a test function name to improve code clarity and maintainability.

- **Refactor**
	- Updated internal logic for merging blocks without style during paste operations.
	- Improved table cell ID creation and parsing using a new custom separator for better consistency and reliability.

### CODE REVIEW

File: core/block/editor/table/table.go [hunk 1]
```diff
@@ -9,6 +9,7 @@ import (
 
 	"github.com/anyproto/anytype-heart/core/block/editor/smartblock"
 	"github.com/anyproto/anytype-heart/core/block/editor/state"
+	"github.com/anyproto/anytype-heart/core/block/editor/table/tableutils"
```
Comment (deff7) [inline core/block/editor/table/table.go:12]: we got simple/table package that has minimal dependencies, maybe use it?

Comment (fat-fellow) [reply core/block/editor/table/table.go]: yep thx

File: core/block/simple/table/utils.go [hunk 2]
```diff
@@ -0,0 +1,9 @@
+package table
+
+import "strings"
+
+const TableCellSeparator = "-"
+
+func IsTableCell(blockId string) bool {
+	return strings.Contains(blockId, TableCellSeparator)
```
Comment (deff7) [inline core/block/simple/table/utils.go:8]: maybe do more strict check? Cells have exactly this shape: `bsonHex-bsonHex`, so check like this should work well:
```
tokens := strings.Split(blockId, separator)
return len(tokens) == 2
```

Comment (fat-fellow) [reply core/block/simple/table/utils.go:8]: > maybe do more strict check
Shall we?
We've had a bug with wrongly generated ids some time before, which includes uuid and IDs with more than two delimiters.
I've decided to to establish a rule where any ID not related to table cells can not contain any delimiters.
If it does, I don't consider it a proper Id, and then all the content will be pasted as if it were in a table cell.
It is never wrong, and even if we somehow encounter an incorrect table cell id with more than 2 delimiters everything will work just fine.

Comment (deff7) [reply core/block/simple/table/utils.go:8]: cell id has special meaning: rowId + columnId

**Review by deff7 — APPROVED** [review]

**Pull request merged**